### DISCUSSION TITLE
errors when update_and_rebuild_libmesh.sh in MOOSE framework

### DISCUSSION THREAD

**Discussion by Joseph-0123** [libMesh/libmesh#3456] [Q&A] [answered]
Dear developers and users,

I am using the [MOOSE framework](https://mooseframework.inl.gov/index.html) on our [uni cluster](https://www.scc.kit.edu/dienste/bwUniCluster_2.0.php) for numerical simulations. Initially, the MOOSE Framework ran well using mpi/openmpi/4.0. After the current maintenance of UniCluster, my MOOSE framework cannot be used due to the [MPI compiler problem](https://github.com/idaholab/moose/discussions/22720). So I use the mpi/openmpi/4.1 to replace the mpi/openmpi/4.0 and update and rebuild MOOSE following their [guidance](https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html) and tips on [mpich buliding](https://mooseframework.inl.gov/getting_started/installation/manual_installation_llvm.html).

I have the following errors when I run ./scripts/update_and_rebuild_libmesh.sh at the ending step of the rebuilding MOOSE.

**/opt/bwhpc/common/mpi/openmpi/4.1.4-gnu-10.2/lib64/libopen-pal.so: undefined reference to fi_open@FABRIC_1.5'`**

Could you please help me? Thanks a lot. 

Best regards,

Joseph

**Comment by jwpeterson** [comment, marked as answer]
It's difficult to comment accurately on your specific cluster, but, generally speaking, we use the mpicxx script to do the compiling and linking. For example, on my system:
```
$ mpicxx -show
g++ -Wl,-Bsymbolic-functions -Wl,-z,relro -I/usr/include/mpich -L/usr/lib/x86_64-linux-gnu -lmpichcxx -lmpich
```
so the first thing you could check is whether you can compile and link a "simple" MPI program using this approach. (Just pick any simple MPI hello_world off the Internet and try compiling and linking it.) This might give us a hint about what to try next...

**Reply by Joseph-0123** [reply]
> It's difficult to comment accurately on your specific cluster, but, generally speaking, we use the mpicxx script to do the compiling and linking. For example, on my system:
> 
> ```
> $ mpicxx -show
> g++ -Wl,-Bsymbolic-functions -Wl,-z,relro -I/usr/include/mpich -L/usr/lib/x86_64-linux-gnu -lmpichcxx -lmpich
> ```
> 
> so the first thing you could check is whether you can compile and link a "simple" MPI program using this approach. (Just pick any simple MPI hello_world off the Internet and try compiling and linking it.) This might give us a hint about what to try next...

Thanks for your suggestions.
